src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/RemoteAdhocWorkspaceFactory.cs

```diff
@@ -2,16 +2,19 @@
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
 using System.Collections.Immutable;
+using System.Composition;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Razor.Workspaces;
 
 namespace Microsoft.CodeAnalysis.Remote.Razor;
 
-internal sealed class RemoteAdhocWorkspaceFactory(HostServices hostServices) : IAdhocWorkspaceFactory
+[Export(typeof(IAdhocWorkspaceFactory)), Shared]
+internal sealed class RemoteAdhocWorkspaceFactory() : IAdhocWorkspaceFactory
 {
     public AdhocWorkspace Create(params IWorkspaceService[] workspaceServices)
     {
         workspaceServices ??= [];
+        var hostServices = RemoteWorkspaceAccessor.GetWorkspace().Services.HostServices;
```

Comment:
Seems like this should be used instead of getting `IWorkspaceService`s from MEF? On the other hand, since I'm guessing `workspaceServices` is empty at runtime, and always will be, maybe we can just remove both and pass in `[]` to the `Create` method?

Reply:
You are right, don't really need MEF here. Removed that.

Would it be safe to always pass empty arrays there? I didn't make that change for now, but can if you feel it's the right thing.

Reply:
In case you missed this one, I think this should have `Export(typeof(IAdhocWorkspaceFactory))` put back, and the direct construction in the remote formatting service should be removed.

Apologies again for the confusion in my initial comments.

Reply:
Saw it, sorry, haven't worked my way down the list yet, coming soon :) 

I'll leave the rest of the code as is for now, as I mentioned if we want to do further cleanup/improvements I'd love to do that is a separate PR instead.